documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
 ...
 ```
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4032"></a>R4032 MissingXmsErrorResponse
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : The response code {0} is defined without a x-ms-error-response.
+
+**Description** :  If defines response code 4xx or 5xx ,  x-ms-error-response:true is required. There is one exception: a HEAD operation with 404 SHOULD have x-ms-error-response:false.
+
+**CreatedAt**: February 23, 2021
+
+**LastModifiedAt**: February 23, 2021
+
+**Why this rule is important**: As some SDK may treat the 4xx or 5xx as exceptional code, if don't specified x-ms-error-response:true, the SDK will not handle the error schema correctly instead it will throw an exception.
+
+**How to fix the violation**: Add the x-ms-error-response:true for the error response code.
+
+The following would be valid:
+
+```json
+ "responses": {
+    "404": {
+      "description": "Not Found",
+      "x-ms-error-response": true
+    }
+ }
```

Comment:
seems conflict with the description:

There is one exception: a HEAD operation with 404 SHOULD have x-ms-error-response:false.

Reply:
fixed